src/hotspot/share/classfile/classListParser.cpp

```diff
@@ -460,19 +461,46 @@ bool ClassListParser::is_matching_cp_entry(constantPoolHandle &pool, int cp_inde
   }
   return true;
 }
-
 void ClassListParser::resolve_indy(Symbol* class_name_symbol, TRAPS) {
+  ClassListParser::resolve_indy_impl(class_name_symbol, THREAD);
+  if (HAS_PENDING_EXCEPTION) {
+    LogTarget(Debug, cds, lambda) log;
+    if (log.is_enabled()) {
+      ResourceMark rm(THREAD);
+      LogStream ls(log);
+      log.print("resolve_indy for class %s has", class_name_symbol->as_C_string());
+      oop message = java_lang_Throwable::message(PENDING_EXCEPTION);
+      if (message != NULL) {
+        char* ex_msg = java_lang_String::as_utf8_string(message);
+        ls.print(" exception pending '%s %s'",
+                  PENDING_EXCEPTION->klass()->external_name(), ex_msg);
+      } else {
+        ls.print(" exception pending %s ",
+                  PENDING_EXCEPTION->klass()->external_name());
+      }
+    }
+    oop exception = THREAD->pending_exception();
+    if (exception->is_a(SystemDictionary::LinkageError_klass())) {
+      // Clear the exception associated with errors like UnsupportedClassVersionError
+      // or NoSuchMethodError so that CDS dumping can continue.
+      CLEAR_PENDING_EXCEPTION;
+    } else {
+      exit(1);
```

Comment:
We should never just use `exit()` - at a minimum `vm_exit()`. But we should not silently exit with no indication of what went wrong. I'm not sure of the call chain here but can't we just return and let higher layers do the same until we eventually hit a cleaner exit path that reports the exception?

Edit: I see now that this code was moved from another location not introduced with this fix, but this still seems wrong to me.

Reply:
For now, I think the logging in ClassListParser::resolve_indy should just by tty->print instead of using UL. This way, if the exit(1) is hit, there will be some information about the reason. Also, not too many users would enable -Xlog:cds+lambda=debug.